examples_tests/06.MeshLoaders/main.cpp

```diff
@@ -150,59 +326,41 @@ int main()
                     {
                         core::matrix4SIMD mvp = camera->getConcatenatedMatrix();
                         memcpy(uboData.data()+shdrIn.descriptorSection.uniformBufferObject.relByteoffset, mvp.pointer(), shdrIn.descriptorSection.uniformBufferObject.bytesize);
-                    }
-                    break;
+                    } break;
+                    
                     case asset::IRenderpassIndependentPipelineMetadata::ECSI_WORLD_VIEW:
                     {
                         core::matrix3x4SIMD MV = camera->getViewMatrix();
                         memcpy(uboData.data() + shdrIn.descriptorSection.uniformBufferObject.relByteoffset, MV.pointer(), shdrIn.descriptorSection.uniformBufferObject.bytesize);
-                    }
-                    break;
+                    } break;
+                    
                     case asset::IRenderpassIndependentPipelineMetadata::ECSI_WORLD_VIEW_INVERSE_TRANSPOSE:
                     {
                         core::matrix3x4SIMD MV = camera->getViewMatrix();
                         memcpy(uboData.data()+shdrIn.descriptorSection.uniformBufferObject.relByteoffset, MV.pointer(), shdrIn.descriptorSection.uniformBufferObject.bytesize);
-                    }
-                    break;
+                    } break;
                 }
             }
         }       
-        driver->updateBufferRangeViaStagingBuffer(gpuubo.get(), 0ull, gpuubo->getSize(), uboData.data());
+
+        commandBuffer[0]->updateBuffer(gpuubo.get(), 0ull, gpuubo->getSize(), uboData.data());
 
         for (auto gpumb : gpumesh->getMeshBuffers())
         {
-            const video::IGPURenderpassIndependentPipeline* pipeline = gpumb->getPipeline();
+            const video::IGPURenderpassIndependentPipeline* gpuRenderpassIndependentPipeline = gpumb->getPipeline();
             const video::IGPUDescriptorSet* ds3 = gpumb->getAttachedDescriptorSet();
+            
+            commandBuffer[0]->bindGraphicsPipeline(gpuGraphicsPipeline.get()); // I'm wondering if I should create gpuGraphicsPipeline for each mesh buffer before
```

Comment:
> I'm wondering if I should create gpuGraphicsPipeline for each mesh buffer before

yes, ofc, as many graphics pipelines as there is unique renderpass independent pipelines

Reply:
well, in that case I will update the code